models/data_maps.go

```diff
@@ -150,6 +160,11 @@ func (d *DataMap) DecryptEthKey(encryptedKey string) (string, error) {
 	return hex.EncodeToString(decryptedKey), nil
 }
 
+func (d *DataMap) generateMsgId() string {
```

Comment:
I don't like this because it require you to save DataMap first before getting this key. This requires SQL operation. For example, if you need to get a particular data, via Genesis and chunkID, it is very fast to compute it. But now, you had to do a look up.

Reply:
This is to ensure that a dataMap will always have a `MsgID` before being saved. This is safer than always generating the key incase the way to generate a key is changed. Also the key for a datamap should be immutable once it's created, this is better documented as a field in the DB.

AFAIK, you will always have a datamap before using the binary data. I don't think we access the binary data outside of the context of a datamap.